docs/superpowers/specs/2026-04-13-e2e-test-isolation-design.md

```
@@ -0,0 +1,103 @@
+
+E2E test scenarios previously shared cluster-scoped resources (ClusterCatalogs, CRDs, packages),
+causing cascading failures when one scenario left state behind. Parallelism was impossible because
+scenarios conflicted on shared resource names.
+
+## Solution
+
+Each scenario dynamically builds and pushes its own bundle and catalog OCI images at test time,
+parameterized by scenario ID. All cluster-scoped resource names include the scenario ID, making
+conflicts structurally impossible.
+
+```
```

markdownlint-cli2 (0.22.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)